xarray/core/options.py

```diff
@@ -1,9 +1,39 @@
 from __future__ import absolute_import, division, print_function
 
+DISPLAY_WIDTH = 'display_width'
+ARITHMETIC_JOIN = 'arithmetic_join'
+ENABLE_CFTIMEINDEX = 'enable_cftimeindex'
+FILE_CACHE_MAXSIZE = 'file_cache_maxsize'
+
 OPTIONS = {
-    'display_width': 80,
-    'arithmetic_join': 'inner',
-    'enable_cftimeindex': False
+    DISPLAY_WIDTH: 80,
+    ARITHMETIC_JOIN: 'inner',
+    ENABLE_CFTIMEINDEX: False,
+    FILE_CACHE_MAXSIZE: 512,
```

Comment:
Is this a good default limit for the max number of files opened by xarray? Maybe a smaller number would be safer?

I picked this number as half of 1024, which is a file descriptor limit I've encountered before. I don't really know what the range of limits encountered in the wild is. My Googling isn't turning much up.

Reply:
I think macOS has the lowest default limit of the OSes, but could be wrong. On macOS the default is 256. Though users can change this with `ulimit`.

Maybe checking [`resource.RLIMIT_NOFILE`]( https://docs.python.org/3/library/resource.html#resource.RLIMIT_NOFILE ) would be appropriate?

Reply:
Hmm, in that case I might just do 128. A fixed number has the virtue of predictability, but `resource.RLIMT_NOFILE // 2` has the virtue of rarely giving users any reason for manual intervention...

Reply:
That's probably reasonable.

Did some searching after that comment and was unable to confirm `resource.RLIMT_NOFILE` works on Windows. It's a UNIX thing historically, but sometimes Python implements operations like these to work on Windows as well.

Should add the Python documentation doesn't make this very clear, but usage would be `resource.getrlimit(resource.RLIMIT_NOFILE)[0]` (for the minimum).

Reply:
I suggest we proceed with something like 128 and then add additional complexity later - provided there is interest.